examples/cmd/main.go

```diff
@@ -147,39 +210,82 @@ func (c *DownloaderCmd) ExecuteBody(logger log.Logger, fn bodyFunc) error {
 		if err = database.Migrate(c.DB); err != nil && err != migrate.ErrNoChange {
 			return err
 		}
-
-		downloader, err = github.NewDownloader(client, db)
 	}
 
-	rate0, err := downloader.RateRemaining(ctx)
+	downloadersPool, err := c.buildDownloadersPool(logger, db)
 	if err != nil {
 		return err
 	}
-	t0 := time.Now()
 
-	err = fn(logger, client, downloader)
+	err = downloadersPool.Begin(ctx)
 	if err != nil {
 		return err
 	}
 
-	err = downloader.SetCurrent(ctx, c.Version)
+	err = fn(logger, downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	if c.Cleanup {
-		return downloader.Cleanup(ctx, c.Version)
-	}
-
-	elapsed := time.Since(t0)
-
-	rate1, err := downloader.RateRemaining(ctx)
+	err = c.commit(ctx, downloadersPool)
 	if err != nil {
 		return err
 	}
-	rateUsed := rate0 - rate1
 
-	logger.With(log.Fields{"rate-limit-used": rateUsed, "total-elapsed": elapsed}).Infof("All metadata fetched")
+	stats, err := downloadersPool.End(ctx)
+
+	logger = logger.With(log.Fields{"total-elapsed": stats.Elapsed})
+	logger.Infof("all metadata fetched")
+	for _, ru := range stats.RatesUsage {
+		logger.With(log.Fields{
+			"rate-limit-used":  ru.Used,
+			"rate-usage-speed": fmt.Sprintf("%f/min", ru.Speed),
+		}).Infof("token usage")
```

Comment:
didn't run the code but looks like `total-elapsed` will be present here. better show it only with `all metadata fetched`